docs/docs/how-to/plugins-and-themes/adding-gatsby-image-support.md

```diff
@@ -0,0 +1,261 @@
+    Omit<GatsbyImageProps, "image"> {
+  // Any other props can go here
+  myCustomProp?: string
+}
+```
+
+Your component can accept just a URL if that's enough to identify the image, or you could pass a full object. This could be an object that you have passed in through your GraphQL API, or it could be data coming from outside of Gatsby, such as via search results or a shopping cart API. Unlike the GraphQL resolvers or the built-in `StaticImage` component, this can be dynamic data that changes at runtime.
+
+For best results you should pass in the dimensions of the source image, so if this data is available then you should ensure it is included. This means the plugin can calculate aspect ratio, and the maximum size image to request.
+
+The component itself should just wrap `<GatsbyImage>`, using your image data function to generate the object to pass to it.
+
+```jsx
+import * as React from "react"
+import { GatsbyImage } from "gatsby-plugin-image"
+import { getExampleImageData } from "./my-image-data"
+
+export function ExampleImage({
+  // Destructure the props that you are passing to the image data function
+  image,
+  width,
+  height,
+  layout,
+  backgroundColor,
+  sizes,
+  aspectRatio,
+  options,
+  // Use ...rest for the GatsbyImage props
+  ...props
+}) {
+  const imageData = getExampleImageData({
+    image,
+    width,
+    height,
+    layout,
+    backgroundColor,
+    sizes,
+    aspectRatio,
+    options,
+  })
+
+  // Pass the image data and spread the rest of the props
+  return <GatsbyImage image={imageData} {...props} />
+}
+```
+
+The user could then use the component like this:
+
+```jsx
+// This might come from an API at runtime
+const image = {
+  url: "baseurl.jpg",
+  width: 800,
+  height: 600,
+}
+
+return (
+  <ExampleImage
+    image={image}
+    layout="fixed"
+    width={400}
+    backgroundColor="#660033"
+    alt="My image"
+  />
+)
+```
+
+A different component for an image CDN might not expect the user to know the dimensions of the source image, and might want to allow them to just pass a base URL:
+
+```jsx
+<ExampleImage
+  image="https://example.com/nnnnn/bighero.jpg"
+  loading="eager"
+  layout="fullWidth"
+  aspectRatio={16 / 9}
+  alt=""
+/>
+```
+
+## Other considerations
+
+You should add `gatsby-plugin-image` as a peer dependency to your plugin, and tell your users to install it in your docs. Don't add it as a direct dependency, as this could lead to multiple versions being installed. You can refer users to [the `gatsby-plugin-image` docs](https://www.gatsbyjs.com/docs/how-to/images-and-media/using-gatsby-plugin-image/) for instructions on using the component, but be sure to document the specifics of your own resolver: it may not be immediately clear to users that the args are different from those in sharp. You may want to highlight specific differences, such as if you don't support AVIF images, or if you do support GIFs, as well as explaining the placeholders that you support.
```

Comment:
```suggestion
You should add `gatsby-plugin-image` as a peer dependency to your plugin, and tell your users to install it in your docs. Don't add it as a direct dependency, as this could lead to multiple versions being installed. You can refer users to [the `gatsby-plugin-image` docs](https://www.gatsbyjs.com/docs/how-to/images-and-media/using-gatsby-plugin-image/) for instructions on using the component. However, be sure to document the specifics of your own resolver: it may not be immediately clear to users that the args are different from those in sharp. You may want to highlight specific differences, such as if you don't support AVIF images, or if you do support GIFs, as well as explaining the placeholders that you support.
```